### PR TITLE
Register BNS observers with Factory

### PR DESCRIPTION
**Pull request by ffoucart**
## Proposed changes

Modifications needed to use BNS observers within spectre evolutions

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### Further comments

A few modifications where needed in order to make it possible to use the new observers and the integrals of conserved quantities in BNS evolution.

(1) Adding the new observers to the list of observers in GhValenciaDivCleanBase.hpp
(2) Adding the tags that we would like to integrate for diagnostics purposes (e.g. MassWeightedInternalEnergyCompute) to the observe_field list in GhValenciaDivCleanBase.hpp (should we instead have a variable elsewhere in the code with the list of integrands we want to use?)
(3) Because SpatialVelocityOneForm is not at this point a db::ComputeTag and it is not stored in memory, I changed the input of an integrand from "SpatialVelocityOneForm" to "SpatialVelocity" and "SpatialMetric" and modified the relevant tests.
(4) Create an integrand_list variable containing the tags to be used in the VolumeIntegral observer (we can't input all tags, because the current version of VolumeIntegral does not handle optionals... and there are a lot of tensors there whose integral is pretty meaningless anyways)

### CODE REVIEW

**Comment by ffoucart** [comment]
See "Further Comments" in the original message for some open questions about this PR.

File: src/ParallelAlgorithms/Events/Factory.hpp [hunk 1]
```diff
@@ -15,11 +15,11 @@
 namespace dg::Events {
 template <size_t VolumeDim, typename TimeTag, typename Fields,
           typename NonTensorComputeTagsList, typename ArraySectionIdTag = void>
-using field_observations = tmpl::flatten<
-    tmpl::list<ObserveFields<VolumeDim, TimeTag, Fields,
```
Comment (knelli2) [inline src/ParallelAlgorithms/Events/Factory.hpp]: This appears to only be a formatting change. Can you remove this or put it in a separate commit?

**Review by knelli2 — CHANGES_REQUESTED** [review]
You can squash this in directly.

I really don't see what could make this time out for just one of our CI builds. Has it been timing out on the CI on your fork as well? When you squash, rebase on latest develop and lets see if it happens a 3rd time.

Edit: Somebody else should review to answer the open questions in the description. I just glanced at things to try and figure out why CI timed out and just noticed one thing.

File: src/ParallelAlgorithms/Events/Factory.hpp (hunk 1 above)
Comment (ffoucart) [reply src/ParallelAlgorithms/Events/Factory.hpp]: Yes, just a remnant of the first method I tried to implement this... I squashed the change, as it doesn't do anything.

**Comment by ffoucart** [comment]
((Latest version is rebased against develop 04/04))

File: src/Evolution/Executables/GrMhd/GhValenciaDivClean/GhValenciaDivCleanBase.hpp [hunk 2]
```diff
@@ -427,6 +434,13 @@ struct GhValenciaDivCleanTemplateBase<
               volume_dim, Frame::Inertial>,
           ::Events::Tags::ObserverCoordinatesCompute<volume_dim,
                                                      Frame::Inertial>>>;
+  using integrand_fields = tmpl::push_back<tmpl::append<
```
Comment (nilsvu) [inline src/Evolution/Executables/GrMhd/GhValenciaDivClean/GhValenciaDivCleanBase.hpp]: delete the empty `push_back`

**Review by nilsvu — CHANGES_REQUESTED** [review]
small thing, squash it in

**Comment by ffoucart** [comment]
Fixed and squashed.

**Review by nilsvu — APPROVED** [review]
